Introduce optee_test implementation

oleksiimoisieiev opened the pull request:
Pushing patch series that implements Zephyr port of the optee_test v3.18.0.
This series includes clean history of the patches that apt to be pushed to the main.

dsemenets committed:
optee_test: Moving optee_test infrastructure to zephyr-optee-test

Taken basic includes and code infrastructure from the optee_test suite
[0] repo, version 3.18.0.
Porting optee_test suite to Zephyr as a standalone application helps
in testing the Zephyr port of the optee_client library [1].

[0] https://github.com/OP-TEE/optee_test.gitz
[1] https://github.com/xen-troops/zephyr-optee-client.git

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

dsemenets committed:
adbg: Adaptation of the adbg functionality to ztest

Zephyr has internal API for the unit tests called ztest. Original
optee-test uses it's own ADBG macro to perform testing.
ADBG was modified to be compatible with Zephyr ztest so it can be run
as Zephyr test suite.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
---
Changes
- Add License to CMakefile
- Add license/copyright to h3ulcb overlay

dsemenets committed:
Moving tee_api_types.h from the optee_os

Porting tee_api_types.h file from the original optee_os repo [0]
version 3.18.0. This is basic header with the essential definitions.

[0] https://github.com/OP-TEE/optee_os.git

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

dsemenets committed:
Refactored funcions BeginSubCase and EndSubCase

Make functions with printf-stile arguments
Added Do_ADBG_BeginSubCase Do_ADBG_EndSubCase macros
Added level variable

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
prj.conf: Disabled OPTEE_SHELL support as deprecated config

Optee_client does not support OPTEE_SHELL anymore
Removed CONFIG_SHELL option
Added CONFIG_LIBCKTEEC option

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

dsemenets committed:
pkcs11_1000: Added suite pkcs11_1000

Ported pkcs11_1000 test suite from the official optee_test repo [0]
version 3.18.0.

[0] https://github.com/OP-TEE/optee_test.git

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
include: Moving ext headers from optee_os

Original optee_test is using ext headers from optee_os repo.
This app can't be dependend from optee_os so porting this headers
to the include dir.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
src: nist: Ported nist headers from optee_test

Nist headers are used by different test cases, such as pkcs11.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

dsemenets committed:
boards: Added dts for rcar_spider board

Add support of the zephyr-optee-test to rcar_spider board.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
Introduce xtest_helpers functionality

This files are including crypto helpers functions, needed for tests.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
Implementation of the adbg_log function to match the original api

ADBG_Log is widely used across the test suite

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

sa-kib committed:
prj.conf: increase app memory

In tight memory setup some tests fail:

SUITE FAIL -  30.00% [regression_1000]: pass = 6, fail = 14, skip = 0, total = 20 duration = 2.37
3 seconds
 - PASS - [regression_1000.test_1001] duration = 0.023 seconds
 - FAIL - [regression_1000.test_1002] duration = 0.030 seconds
 - PASS - [regression_1000.test_1003] duration = 0.051 seconds
 - FAIL - [regression_1000.test_1004] duration = 0.284 seconds
 - FAIL - [regression_1000.test_1005] duration = 0.059 seconds
 - FAIL - [regression_1000.test_1006] duration = 0.052 seconds
 - FAIL - [regression_1000.test_1007] duration = 0.050 seconds
 - FAIL - [regression_1000.test_1008] duration = 0.148 seconds
 - FAIL - [regression_1000.test_1009] duration = 0.176 seconds
 - FAIL - [regression_1000.test_1010] duration = 0.558 seconds
 - PASS - [regression_1000.test_1011] duration = 0.033 seconds
 - PASS - [regression_1000.test_1012] duration = 0.039 seconds
 - FAIL - [regression_1000.test_1013] duration = 0.458 seconds
 - FAIL - [regression_1000.test_1016] duration = 0.059 seconds
 - PASS - [regression_1000.test_1017] duration = 0.017 seconds
 - FAIL - [regression_1000.test_1018] duration = 0.053 seconds
 - FAIL - [regression_1000.test_1019] duration = 0.053 seconds
 - PASS - [regression_1000.test_1020] duration = 0.011 seconds
 - FAIL - [regression_1000.test_1021] duration = 0.165 seconds
 - FAIL - [regression_1000.test_1022] duration = 0.054 seconds

With increase memory:

SUITE FAIL -  40.00% [regression_1000]: pass = 8, fail = 12, skip = 0, total = 20 duration = 6.70
3 seconds
 - PASS - [regression_1000.test_1001] duration = 0.023 seconds
 - PASS - [regression_1000.test_1002] duration = 0.006 seconds
 - PASS - [regression_1000.test_1003] duration = 0.051 seconds
 - PASS - [regression_1000.test_1004] duration = 0.055 seconds
 - PASS - [regression_1000.test_1005] duration = 0.019 seconds
 - FAIL - [regression_1000.test_1006] duration = 0.058 seconds
 - FAIL - [regression_1000.test_1007] duration = 0.056 seconds
 - FAIL - [regression_1000.test_1008] duration = 0.165 seconds
 - FAIL - [regression_1000.test_1009] duration = 0.199 seconds
 - FAIL - [regression_1000.test_1010] duration = 0.867 seconds
 - PASS - [regression_1000.test_1011] duration = 0.039 seconds
 - PASS - [regression_1000.test_1012] duration = 0.045 seconds
 - FAIL - [regression_1000.test_1013] duration = 4.622 seconds
 - FAIL - [regression_1000.test_1016] duration = 0.064 seconds
 - FAIL - [regression_1000.test_1017] duration = 0.058 seconds
 - FAIL - [regression_1000.test_1018] duration = 0.058 seconds
 - FAIL - [regression_1000.test_1019] duration = 0.059 seconds
 - PASS - [regression_1000.test_1020] duration = 0.011 seconds
 - FAIL - [regression_1000.test_1021] duration = 0.188 seconds
 - FAIL - [regression_1000.test_1022] duration = 0.060 seconds

Signed-off-by: Sergiy Kibrik <Sergiy_Kibrik@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

sa-kib committed:
Add regession_6000 suite

Port whole regression_6000 suite from optee_test (19 tests), which cover
permanent objects storage. The original suite detects supported storage types
on the system and has every test executed for each of them: REE and/or RPMB,
plus whichever is default. In this port however the storage type is fixed
to REE for now, as we got no RPMB support in Zephyr's MMC driver yet.

Signed-off-by: Sergiy Kibrik <Sergiy_Kibrik@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

sa-kib committed:
Enable LFS storage backed by flash simulator

Use LFS as REE filesystem for regression_6000 suite.

Signed-off-by: Sergiy Kibrik <Sergiy_Kibrik@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

sa-kib committed:
boards: h3ulcb: correct ram range for dom0 + LFS + flash

Put a proper ram range for xen_dom0, also mount LFS on /tee to use as REE
filesystem storage, backed by flash simulator for simplicity.

Signed-off-by: Sergiy Kibrik <Sergiy_Kibrik@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
regression_4100: Introduce regression_4100 test suite

This test_suite was ported from the optee_test repo v3.18.0 and
adopted to zephyr xtext.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
regression_8000: Introducing regression_8000 test suite

This test_suite was ported from optee_test v3.18.0 and adopted to be
used with Zephyr ztest.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
regression_1000: Porting test_suite from optee_test

Ported basic test suite from the optee_test v3.18.0.
This test suite does the basic optee features testing.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
boards: Add flash simulator and lfs support to rcar_spider

Add flash simulator and lfs for secure storage to rcar_spider board overlay.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
LICENSE: Modify license to GPL-v2 compatibility

The original optee_test source code which was ported on Zephyr was
licensed as GPL-v2 which is not compatible with the original Apache-2
license.
Switching license file to match GPL-v2 license.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
Introduce regression_4000 support for optee_test

4000 series is intended to test basic cryptographic functionality

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

firscity commented on adbg/CMakeLists.txt:
```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: GPL-v2
```
Commit message `optee test application: Initial commit`: please, make it according to community standards, mention source and revision.

oleksiimoisieiev replied:
Reworded

firscity commented on adbg/include/adbg.h:
```diff
@@ -0,0 +1,142 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+//#include <sys/queue.h>
```
No commented code.

oleksiimoisieiev replied:
fixed

firscity commented on adbg/CMakeLists.txt:
```diff
@@ -0,0 +1,19 @@
+target_include_directories(OPTEE_TEST_ADBG INTERFACE include)
+
+zephyr_library()
+#zephyr_library_sources(src/adbg_case.c)
```
No commented code.

oleksiimoisieiev replied:
+

firscity commented on adbg/src/adbg_log.c:
```diff
@@ -0,0 +1,56 @@
+		}
+		AsciiBuffer[m] = '\0';
+
+//		LOG_PRINTK("  %-*s %s", (int)Cols * 3, HexBuffer, AsciiBuffer);
```
No commented code.

oleksiimoisieiev replied:
+

firscity commented on src/optee_test.c:
```diff
@@ -0,0 +1,102 @@
+#include <stdlib.h>
```
License

firscity commented on scripts/rsp_to_gcm_test.py:
```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
```
License

firscity commented on src/optee_test.h:
```diff
@@ -0,0 +1,75 @@
+#ifndef __OPTEE_TEST_H
```
License

firscity commented on src/regression_1000.c:
```diff
@@ -0,0 +1,157 @@
+#include <errno.h>
```
License

firscity commented on west.yml:
```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: Apache-2.0
+# West YAML manifest for Zephyr as Xen Dom0
```
Is it true?

oleksiimoisieiev replied:
removed

firscity commented on CMakeLists.txt:
```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.20.0)
```
License

oleksiimoisieiev replied:
+

firscity replied:
Still not here

oleksiimoisieiev replied:
+

firscity commented on prj.conf:
```diff
@@ -2,7 +2,10 @@ CONFIG_ZTEST=y
 CONFIG_OPTEE_CLIENT=y
 CONFIG_OPTEE_LIBTEEC=y
-CONFIG_OPTEE_DRV_STUB=y
+#CONFIG_OPTEE_DRV_STUB=y
```
No commented code

oleksiimoisieiev replied:
+

firscity commented on adbg/src/adbg_expect.c:
```diff
@@ -195,9 +204,9 @@ bool Do_ADBG_ExpectComparePointer(
-		LOG_PRINTK(
-			"%s:%d: Expression \"%s %s %s\" (%p %s %p) is false",
+	if (!ADBG_AssertHelper(c, FileName_p, LineNumber, Result)) {
+		printk(
```
Direct usage of printk is deprecated, it is better to use LOG_INF/_ERR/_DBG

oleksiimoisieiev replied:
This was done similar to z_zassert implementation, which is also using printk under the hood. This was done because macro does formatting. Using LOG_* will make LOG_* formatting + xtest formatting. We want to minimize the difference between zephyr-optee-test and optee_test output which can be handy in future for example it will be easier to compare and parse output in CI (if someone dare make one). Also I'm not sure adbg internals is the right place for LOG_MODULE_REGISTER call.

oleksiimoisieiev replied:
Also ADBG has it's own macro, called ADBG_LOG. I think the implementation of ADBG_Log using LOG_DBG, or replacing ADBG_LOG to LOG_DBG across all project is not needed.

firscity commented on src/optee_test.c:
```diff
@@ -33,6 +34,8 @@ TEEC_Context xtest_teec_ctx;
 
 char *xtest_tee_name = NULL;
 
+int level = 0;
```
static?

oleksiimoisieiev replied:
Thats a bit of ugly solution. But we have what we have. All test suites requires level to be set explicitly. Some of the tests requires level > 0. That's why static can't be used here - some tests may want to set different level.

firscity commented on src/sdp_basic.c:
```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>                                                                                                                                                                                     
```
License

oleksiimoisieiev replied:
+

firscity reviewed (commented):
All commits should have normal description

oleksiimoisieiev commented on src/optee_test.c:
```diff
@@ -90,13 +90,13 @@ void xtest_mutex_lock(struct k_mutex *mutex)
 
 	if (e)
-		printk( "pthread_mutex_unlock: %s", strerror(e));
+		printk( "pthread_mutex_unlock: %si\n", strerror(e));
```
redundand change

oleksiimoisieiev replied:
fixed

dsemenets committed:
scripts: Add scripts for generating table with loadable ta

Added scripts:
 - gen_ta_src.py - generates .c and .S source files which added
       TA lookup table
 - prepare_ta.sh - script called by build system. This script copies
       TA's from external builder e.g. yocto and passed to zephyr
       builder via TA_DEPLOY_DIR variadble and invokes gen_ta_src.py
       script.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

firscity commented on boards/qemu_cortex_a53.overlay:
```diff
@@ -0,0 +1,64 @@
+
```
License/copyright

oleksiimoisieiev replied:
+

firscity commented on boards/rcar_h3ulcb_ca57.overlay:
```diff
@@ -0,0 +1,59 @@
+/delete-node/ &ram;
```
License/copyright

oleksiimoisieiev replied:
+

firscity commented on boards/rcar_spider.overlay:
```diff
@@ -0,0 +1,52 @@
+/ {
```
License/copyright

oleksiimoisieiev replied:
+

firscity commented on scripts/common.mk:
```diff
@@ -0,0 +1,11 @@
+# Makefile script used in various OP-TEE test components
```
License/copyright

oleksiimoisieiev replied:
Wasn't in original file from optee_test, but added/

firscity reviewed (approved):
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`

@lorc, can you take a quick overview look on implementation? It is intended to be used only as testing project.

dsemenets commented on west.yml:
```diff
@@ -0,0 +1,34 @@
+      remote: xentroops
+      revision: master
+
+    - name: littlefs
```
Do we use littlefs or zephyr-xenlib in this application?

oleksiimoisieiev replied:
We definitely use littlefs for secure storage. As for the zehyr-xenlib - I don't know. You've added it :). Probably we don't need but I'll recheck and update

oleksiimoisieiev replied:
updated

dsemenets reviewed (approved):
+1

lorc commented:
Well, I am not familiar with Zephyr test framework enough to make any judgements about correct use.

Testcase themselves are taken from original optee_test. So no sense in reviewing them. Nevertheless, I have a couple of comments:

1. There is no sense in storing prebuilt TAs. TAs are build and linked against particular OP-TEE build, because they use lots of OP-TEE config options. Also, TAs are signed for the particular platform. You can't expect that prebuild TAs will work as intended on any other OP-TEE image. We need to either build TAs against exported OP-TEE TA Dev Kit or tell the user to provide TAs.

2. There is  a "scripts" folder which is not used anywhere.

firscity commented:
Please, check last unresolved comment, it should be replaced with optee-test description
https://github.com/xen-troops/zephyr-optee-test/pull/4#discussion_r1262622728

oleksiimoisieiev commented:
> 



> Well, I am not familiar with Zephyr test framework enough to make any judgements about correct use.
> 
> Testcase themselves are taken from original optee_test. So no sense in reviewing them. Nevertheless, I have a couple of comments:
> 
>     1. There is no sense in storing prebuilt TAs. TAs are build and linked against particular OP-TEE build, because they use lots of OP-TEE config options. Also, TAs are signed for the particular platform. You can't expect that prebuild TAs will work as intended on any other OP-TEE image. We need to either build TAs against exported OP-TEE TA Dev Kit or tell the user to provide TAs.
> 
>     2. There is  a "scripts" folder which is not used anywhere.

There is no obvious way to provide such TA to the optee test. Correct approach will be to make externalProject in CMake to build optee and then generate ta_table using scripts from the ./scripts folder. Originally it was done the same way it was supposed to be intergrated to aos-core-zephyr application (which lately was transformed to external TAs, provided with env variable). I think the contents of ./scripts directory should be documented so user can generate it's own ta_table, but current TAs are usable for the use cases we have for now. @dsemenets Do you have some Readme with the description of the ./scripts contents? And could you please share your thoughts about that?

lorc commented:
> There is no obvious way to provide such TA to the optee test. 

I agree. I am just pointing out that there is no sense in embedding some prebuild TAs in the test app repo. It is not **guaranteed** that those TAs will work somewhere except committer's local setup. 4

> contents of ./scripts directory

I was talking about those files:

`scripts/checkpatch.sh`
`scripts/common.mk`
`scripts/file_to_c.py`
`scripts/rsp_to_gcm_test.py`

They are not referenced by any other parts of the code:

```
$ for x in `ls scripts`; do echo "$x:"; grep -rni $x * ;done
checkpatch.sh:
scripts/checkpatch.sh:7:source $OPTEE_OS_PATH/scripts/checkpatch.sh "$@"
common.mk:
file_to_c.py:
rsp_to_gcm_test.py:
```

So, why they are included in the commit? 

Could you please reviewed your own changes to ensure that you are not adding any dead code, like those files in the `scripts` directory?

oleksiimoisieiev commented:
> I agree. I am just pointing out that there is no sense in embedding some prebuild TAs in the test app repo. It is not **guaranteed** that those TAs will work somewhere except committer's local setup. 4

After some discussions it was decided to move an approach from aos_core_zephyr  for this [PR](https://github.com/aoscloud/aos_core_zephyr/pull/83). I will move this changes after merging provided PR to aos_core_zephyr

xakep-amatop commented on ta_table/ta_table.c:
```diff
@@ -0,0 +1,251 @@
+	table[31].ta_size = __ta_fd02c9da_306c_48c7_a49c_bbd827ae86ee_size;
+	table[32].ta_start = __ta_ffd2bded_ab7d_4988_95ee_e4962fff7154_start;
+	table[32].ta_size = __ta_ffd2bded_ab7d_4988_95ee_e4962fff7154_size;
+	printk("====== %s %d\n", __func__, __LINE__);
```
do we need this print?

oleksiimoisieiev replied:
refactored

xakep-amatop commented on ta_table/ta_table.c:
```diff
@@ -0,0 +1,251 @@
+/* Do not modify. This is generated file*/
```
Do we need to generate a license preamble?

oleksiimoisieiev replied:
refactored

xakep-amatop commented on ta_table/ta_table.S:
```diff
@@ -0,0 +1,267 @@
+/* Do not modify. This is generated file*/
```
is it generated every time when we compile it? if yes, maybe better to such files to git ignore list?

xakep-amatop commented on src/sdp_basic.h:
```diff
@@ -0,0 +1,51 @@
+
+#include <linux/dma-buf.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
```
is it compiles? if yes how where we have `linux` dirs?

oleksiimoisieiev replied:
removed file

xakep-amatop commented on CMakeLists.txt:
```diff
@@ -0,0 +1,37 @@
+
+zephyr_library_include_directories(${ZEPHYR_OPTEELIB_MODULE_DIR}/public)
+
+#add_subdirectory(adbg)
```
if it isn't needed, just drop this line

oleksiimoisieiev replied:
+

xakep-amatop commented on adbg/include/adbg.h:
```diff
@@ -0,0 +1,165 @@
+bool Do_ADBG_ExpectComparePointer(struct ADBG_Case *c, const char *const FileName_p,
+				  const int LineNumber,
+				  const void *const Value1_p,
+				  const void *const Value2_p, const bool Result,
```
some lines contains 2 parameters per line, some 1 parameter

oleksiimoisieiev replied:
This is original code style. I think there is no reason to update original code style

xakep-amatop commented on adbg/src/adbg_expect.c:
```diff
@@ -0,0 +1,254 @@
+	const char *const GotVarName_p
+	)
+{
+	if (ADBG_AssertHelper(c, FileName_p, LineNumber, Got_p != NULL))
```
should you follow MISRA rules? if yes, we have a lot of issues, for instance, brackets are missed here

oleksiimoisieiev replied:
I think not. This is optee-test application under GPL-v2 license. I don't think it requires certification

xakep-amatop commented on adbg/src/adbg_expect.c:
```diff
@@ -0,0 +1,254 @@
+	const char *Ch_p = FileName_p;
+	const char *Base_p = FileName_p;
+
+	while (*Ch_p != '\0') {
```
look like you can use here `strchr` function

oleksiimoisieiev replied:
Yes I can, But this is the original code. I see no point refactoring it.

xakep-amatop commented on adbg/src/adbg_log.c:
```diff
@@ -0,0 +1,65 @@
+}
+
+/*************************************************************************
+ * 6. Definitions of internal functions
```
where are there other points?

oleksiimoisieiev replied:
Ditto.

xakep-amatop commented on adbg/src/adbg_log.c:
```diff
@@ -0,0 +1,65 @@
+	printk("\n");
+	va_end(ap);
+}
+
```
should you add description for this function too?

oleksiimoisieiev replied:
Ditto.

xakep-amatop commented on adbg/src/security_utils_hex.c:
```diff
@@ -0,0 +1,239 @@
+* 5. Definition of external functions
+*************************************************************************/
+#if 0
+void SecBase_HexPrintBuffer(
```
why do we need dead code here and below?

oleksiimoisieiev replied:
This is original code

xakep-amatop commented on adbg/src/security_utils_hex.h:
```diff
@@ -0,0 +1,118 @@
+* 2. Types, constants and external variables
+*************************************************************************/
+
+#if 0 /* Should format into a buffer instead */
```
should you drop these changes too?

oleksiimoisieiev replied:
Ditto.

xakep-amatop commented on boards/rcar_h3ulcb_ca57.overlay:
```diff
@@ -0,0 +1,65 @@
+		};
+	};
+
+	ram: memory@70000000 {
```
do we really need here this node?

oleksiimoisieiev replied:
Sure we do. App doesn't work on xen otherwise.

xakep-amatop commented on include/enc_fs_key_manager_test.h:
```diff
@@ -0,0 +1,17 @@
+   the ITU-T UUID generator at http://www.itu.int/ITU-T/asn1/uuid.html */
+#define ENC_FS_KEY_MANAGER_TEST_UUID \
+		{ 0x17E5E280, 0xD12E, 0x11E4,  \
+			{ 0xA4, 0x1A, 0x00, 0x02, 0xA5, 0xD5, 0xC5, 0x1B } }
```
Note: It is possible to have lines with 100 symbols.

oleksiimoisieiev replied:
This is the original code

xakep-amatop commented on include/enc_fs_key_manager_test.h:
```diff
@@ -0,0 +1,17 @@
+#ifndef ENC_FS_KEY_MANAGER_TEST_H
+#define ENC_FS_KEY_MANAGER_TEST_H
+
+/* This UUID is generated with uuidgen
```
The style of multiline comments is different in Zephyr.

oleksiimoisieiev replied:
Ditto.

xakep-amatop commented on include/ta_os_test.h:
```diff
@@ -0,0 +1,50 @@
+#define TA_OS_TEST_CMD_PAUTH_NOP            30
+#define TA_OS_TEST_CMD_PAUTH_CORRUPT_PAC    31
+#define TA_OS_TEST_CMD_ATTESTATION          32
+#define TA_OS_TEST_CMD_MEMTAG_USE_AFTER_FREE 33
```
looks like alignment is broken

oleksiimoisieiev replied:
Ditto

xakep-amatop commented on include/ta_socket.h:
is this file used and others? how have you enabled network subsystem in Zephyr?

oleksiimoisieiev replied:
optee_test.c uses certain defines from this file. This happens when you porting code from the different system. We've tried to minimize original code changes

xakep-amatop commented on include/types_ext.h:
```diff
@@ -0,0 +1,36 @@
+/* typedef uintptr_t vaddr_t; */
+#define PRIxVA	PRIxPTR
+
+#if defined(__ILP32__) && defined(CFG_CORE_LARGE_PHYS_ADDR)
```
should you use zephyr defines for that?

oleksiimoisieiev replied:
I would rather leave this configs. This will help during build when you use the same configs for optee_os and optee_test

xakep-amatop commented on scripts/checkpatch.sh:
```diff
@@ -0,0 +1,7 @@
+   OPTEE_OS_PATH="../optee_os";
+fi
+
+source $OPTEE_OS_PATH/scripts/checkpatch.sh "$@"
```
should we use Zephyr checkpatch?

oleksiimoisieiev replied:
This file should be removed

xakep-amatop commented on src/xtest_helpers.h:
```diff
@@ -0,0 +1,103 @@
+ * Helpers for commands towards the crypt TA
+ */
+TEEC_Result ta_crypt_cmd_allocate_operation(struct ADBG_Case *c,
+			   TEEC_Session *s,
```
strange alignment here and below

oleksiimoisieiev replied:
This is the original code style

xakep-amatop commented on include/tee_api_types.h:
```diff
@@ -0,0 +1,204 @@
+*/
+typedef union {
+	struct {
+		void *buffer;
```
I hope that a, b shouldn't be aligned with buffer, size.
In the case of 64-bit pointers, buffer will have a length of 8 bytes.

oleksiimoisieiev replied:
Same as below

xakep-amatop commented on include/tee_api_types.h:
```diff
@@ -0,0 +1,204 @@
+	uint32_t attributeID;
+	union {
+		struct {
+			void *buffer;
```
I hope that a, b shouldn't be aligned with buffer, size.
In the case of 64-bit pointers, buffer will have a length of 8 bytes.

oleksiimoisieiev replied:
This is open header from the optee_os. I don't think there is bug here.

xakep-amatop commented on boards/qemu_cortex_a53.overlay:
```diff
@@ -61,7 +61,7 @@
 	firmware {
 		tee {
-			compatible = "vnd,tee";
+			compatible = "linaro,optee-tz";
```
move this change to the commit 8e4b73a16bdd1ebdbe214ec8d1764c3d63abf2ea

oleksiimoisieiev replied:
Removed this config. We can't start optee with zephyr on qemu anyway, (Because of uefi)

xakep-amatop commented on src/pkcs11_1000.c:
```diff
@@ -0,0 +1,7971 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2018, Linaro Limited
+ * Copyright (c) 2023, EPAM Systems
```
it is sad it isn't possible to view changes that were added by you inside this pull request
the same story about other ported files

oleksiimoisieiev replied:
What do you mean under term "view changes"? If you expect to see original file as a separate commit - that's not a good idea because it will make optee-test unbuildable on this commit. This file, similar to others, was adapted to work with ztest. No functional changes were done

oleksiimoisieiev commented:
> I was talking about those files:
> 
> `scripts/checkpatch.sh` `scripts/common.mk` `scripts/file_to_c.py` `scripts/rsp_to_gcm_test.py`
> 
> They are not referenced by any other parts of the code:
> 
> ```
> $ for x in `ls scripts`; do echo "$x:"; grep -rni $x * ;done
> checkpatch.sh:
> scripts/checkpatch.sh:7:source $OPTEE_OS_PATH/scripts/checkpatch.sh "$@"
> common.mk:
> file_to_c.py:
> rsp_to_gcm_test.py:
> ```
> 
> So, why they are included in the commit?
> 
> Could you please reviewed your own changes to ensure that you are not adding any dead code, like those files in the `scripts` directory?

Completely refactored the code. Removed prebuilt TAs from the build and created a rule to generate ta_table. Also got rid from useless scripts.

firscity reviewed (approved):
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`

lorc commented on scripts/gen_ta_src.py:
```diff
@@ -0,0 +1,142 @@
+                uuids.append(get_components(line))
+    return uuids
+
+def get_components(uuid):
```
There is a  built-in  python module `uuid`:  https://docs.python.org/3/library/uuid.html

No need to use own hacky solutions

oleksiimoisieiev replied:
Refactored in fixup. Used uuid library and have done some optimizations. Made a fixup so it would be easier to see the differences. @dsemenets you may be interested as well.

lorc commented on scripts/gen_ta_src.py:
```diff
@@ -0,0 +1,142 @@
+        i += 1
+    return items;
+
+def compare_uuids(uuid1, uuid2):
```
Again. Just `import uuid` and use the provided API

oleksiimoisieiev replied:
updated

lorc commented on scripts/gen_ta_src.py:
```diff
@@ -0,0 +1,142 @@
+                        if compare_uuids(i, components):
+                            found = True
+                            break;
+                    if not found:
```
You can use `else:` in `for` loop. This will allow you to get rid of `found` variable

oleksiimoisieiev replied:
This was refactored. Got rid from for loop

lorc commented on CMakeLists.txt:
```diff
@@ -26,3 +26,24 @@ zephyr_library_sources(src/regression_6000.c)
+if(TA_DEPLOY_DIR)
+    set(TA_BIN_DIRECTORY ta_bin)
+    set(GENERATED_DIRECTORY ta_table)
+    execute_process(
```
I am not sure I got how it is supposed to work. Main goal of a build system is to track changes in the source files and rebuild all dependent files. In this case, CMake will not track if any of the TAs were rebuilt and will not regenerate `ta_table.c`. Unless I missed something, of course

dsemenets replied:
This way doesn't need enumerate necessary TAs in yaml and CMake. Also it simplified CMakeLists.txt. Regeneration and compilation `ta_table.*` files not a big deal and not significantly affects to overall build time.

lorc replied:
AFAIK, it would be done only during the first build, when you call `cmake` to create `Makefile`/`build.ninja`. If I understand how CMake works, all the following builds will not regenerate `ta_table.*` files:

> execute_process runs commands while CMake is configuring the project, prior to build system generation. Use the [add_custom_target()](https://cmake.org/cmake/help/latest/command/add_custom_target.html#command:add_custom_target) and [add_custom_command()](https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command) commands to create custom commands that run at build time

(https://cmake.org/cmake/help/latest/command/execute_process.html)

oleksiimoisieiev replied:
Just did some tests by calling
`west build -b rcar_h3ulcb_ca57 -p always -- -DTA_DEPLOY_DIR=$(pwd)/prebuilt`
several times. And I can confirm that ta_table is regenerating each time.

lorc replied:
> west build -b rcar_h3ulcb_ca57 -p always -- -DTA_DEPLOY_DIR=$(pwd)/prebuilt

Because `-p always` triggers full rebuild every time.

oleksiimoisieiev replied:
fixed

lorc commented:
Not ideal, but good enough for time being:

`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

oleksiimoisieiev commented:
updated
